src/Observer.hpp

```diff
@@ -19,6 +20,7 @@ class ISubject
   virtual void attach(IObserver* observer) = 0;
   virtual void detach(IObserver* observer) = 0;
   virtual void notify() = 0;
+  virtual ~ISubject(){};
 };
 
 #endif
```

Comment:
I'm surprised clang-format does not complain about the lack of newline at the end of this file!